Examples/src/main/kotlin/com/mapbox/vision/examples/POIActivityKt.kt

```diff
@@ -0,0 +1,262 @@
+            POIState(it, distanceToVehicle, worldCoordinate)
+        }
+
+        // Show only POI which is close enough and behind the car
+        private fun filterPOIStateListToShow(poiStateList: List<POIState>) = poiStateList.filter {
+            val x = it.worldCoordinate.x
+            // Check if POI is behind vehicle and close enough to start appearing
+            (x > 0) && (it.distanceToVehicle < DRAW_LABEL_MIN_DISTANCE_METERS)
+        }
+
+        private fun preparePOIDrawData(poiStateList: List<POIState>): List<POIDrawData> {
+            val poiDrawDataList = mutableListOf<POIDrawData>()
+            for (poiState in poiStateList) {
+                // Prepare bounding rect for POI in mobile screen coordinates
+                val poiBitmapRect = calculatePOIScreenRect(poiState.worldCoordinate)
+                val poiLabelAlpha = calculatePOILabelAlpha(poiState)
+                val poiDrawData = POIDrawData(poiState.poi.bitmap, poiBitmapRect, poiLabelAlpha)
+                poiDrawDataList.add(poiDrawData)
+            }
+            return poiDrawDataList
+        }
+
+        private fun calculatePOIScreenRect(poiWorldCoordinate: WorldCoordinate): Rect {
+            // Calculate left top coordinate of POI in real world using POI world coordinate
+            val worldLeftTop = poiWorldCoordinate.copy(
+                    y = poiWorldCoordinate.y + LABEL_SIZE_METERS / 2,
+                    z = poiWorldCoordinate.z + LABEL_ABOVE_GROUND_METERS + LABEL_SIZE_METERS
+            )
+
+            // Calculate right bottom coordinate of POI in real world using POI world coordinate
+            val worldRightBottom = poiWorldCoordinate.copy(
+                    y = poiWorldCoordinate.y - LABEL_SIZE_METERS / 2,
+                    z = poiWorldCoordinate.z + LABEL_ABOVE_GROUND_METERS
+            )
+            val poiBitmapRect = Rect(0, 0, 0, 0)
+
+            // Calculate POI left top position on camera frame from real word coordinates
+            VisionReplayManager.worldToPixel(worldLeftTop)?.run {
+                poiBitmapRect.left = x
+                poiBitmapRect.top = y
+            }
+
+            // Calculate POI right bottom position on camera frame from real word coordinates
+            VisionReplayManager.worldToPixel(worldRightBottom)?.run {
+                poiBitmapRect.right = x
+                poiBitmapRect.bottom = y
+            }
+            return poiBitmapRect
+        }
+
+        private fun calculatePOILabelAlpha(poiState: POIState): Int {
+            val minDistance = min(DRAW_LABEL_MIN_DISTANCE_METERS - poiState.distanceToVehicle, DISTANCE_FOR_ALPHA_APPEAR_METERS)
+            return ((minDistance / DISTANCE_FOR_ALPHA_APPEAR_METERS.toFloat()) * 255).toInt()
+        }
+
+        private fun drawPOIList(poiDrawDataList: List<POIDrawData>): Bitmap {
+            val bitmap = Bitmap.createBitmap(bitmapCameraFrame)
```

Comment:
Why create new `bitmap` here? We could pass `bitmapCameraFrame` to `Canvas` and just clear `Canvas` with `drawColor(TRANSPARENT)`.

Reply:
good idea

Reply:
fixed